File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/RouteRetrievalEvent.java
package com.mapbox.services.android.navigation.v5.internal.navigation;

import android.annotation.SuppressLint;
import android.os.Parcelable;

@SuppressLint("ParcelCreator")
@SuppressWarnings("ParcelableCreator")
class RouteRetrievalEvent extends NavigationPerformanceEvent implements Parcelable {
  private static final String ELAPSED_TIME_NAME = "elapsed_time";
  private static final String ROUTE_UUID_NAME = "route_uuid";
  private static final String ROUTE_RETRIEVAL_EVENT_NAME = "route_retrieval_event";

  RouteRetrievalEvent(double elapsedTime, String routeUuid, String sessionId,
                      NavigationPerformanceMetadata metadata) {
    super(sessionId, ROUTE_RETRIEVAL_EVENT_NAME, metadata);

    addCounter(new DoubleCounter(ELAPSED_TIME_NAME, elapsedTime));
    addAttribute(new Attribute(ROUTE_UUID_NAME, routeUuid));
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/RouteRetrievalEvent.kt
package com.mapbox.services.android.navigation.v5.internal.navigation

import android.annotation.SuppressLint
import android.os.Parcelable
import com.google.gson.Gson
import com.mapbox.services.android.navigation.v5.internal.navigation.metrics.DirectionsMetrics
import com.mapbox.services.android.navigation.v5.internal.navigation.metrics.NavigationPerformanceEvent
import com.mapbox.services.android.navigation.v5.navigation.metrics.MetricEvent

@SuppressLint("ParcelCreator")
internal data class RouteRetrievalEvent(
    private val elapsedTime: Double,
    private val routeUuid: String,
    @Transient private val sessionId: String,
    @Transient override var metadata: NavigationPerformanceMetadata
) : NavigationPerformanceEvent(sessionId, DirectionsMetrics.ROUTE_RETRIEVAL, metadata), MetricEvent, Parcelable {

    companion object {
        private const val ELAPSED_TIME_NAME = "elapsed_time"
        private const val ROUTE_UUID_NAME = "route_uuid"
    }

    init {
        addCounter(DoubleCounter(ELAPSED_TIME_NAME, elapsedTime))
        addAttribute(Attribute(ROUTE_UUID_NAME, routeUuid))
    }

    override fun toJson(gson: Gson): String = gson.toJson(this)

    override val metric: String
        get() = DirectionsMetrics.ROUTE_RETRIEVAL
}
